pytorch/containers/hp_tuning/task.py

```diff
@@ -0,0 +1,159 @@
+
+    for batch_index, data in enumerate(train_loader):
+        features = data['features']
+        target = data['target']
+
+        # zero the parameter gradients
+        optimizer.zero_grad()
+        # forward + backward + optimize
+        outputs = net(features)
+        loss = criterion(outputs, target)
+        loss.backward()
+        optimizer.step()
+
+
+def set_value_(item):
+    """Take the predictions and assign them a binary target value."""
+    if item < 0.5:
+        return 0.0
+    else:
+        return 1.0
+
+
+def test(net, test_loader):
+    """Test the DNN"""
+    net.eval()
+    criterion = nn.BCELoss()
+    test_loss = 0
+    correct = 0
+
+    with torch.no_grad():
+        for i, data in enumerate(test_loader, 0):
+            features = data['features']
+            target = data['target']
+            output = net(features)
+            # Binarize the output
+            pred = output.apply_(set_value_)
+
+            test_loss += criterion(output, target)  # sum up batch loss
+
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= len(test_loader.dataset)
+    total = len(test_loader) * test_loader.batch_size
+    accuracy = 100. * correct / total
+    return accuracy
+
+
+def task(args):
+    """Load the data, train the model, test the model, export / save the model
+    """
+    torch.manual_seed(args.seed)
+
+    # Open our dataset
+    train_loader, test_loader = data_utils.load_data(
+        args.test_split, args.seed, args.batch_size)
+
+    # Create the model
+    net = model.SonarDNN().double()
+    optimizer = optim.SGD(net.parameters(), lr=args.lr,
+                          momentum=args.momentum, nesterov=False)
+
+    # Train / Test the model
+    latest_accuracy = 0.0
+    for epoch in range(1, args.epochs + 1):
+        train(net, train_loader, optimizer)
+        latest_accuracy = test(net, test_loader)
+
+    # The default name of the metric is training/hptuning/metric.
+    # We recommend that you assign a custom name. The only functional
+    # difference is that if you use a custom name, you must set the
+    # hyperparameterMetricTag value in the HyperparameterSpec object in your
+    # job request to match your chosen name.
+    # https://cloud.google.com/ml-engine/reference/rest/v1/projects.jobs#HyperparameterSpec
+    hpt = hypertune.HyperTune()
+    hpt.report_hyperparameter_tuning_metric(
+        hyperparameter_metric_tag='my_metric_tag',
+        metric_value=latest_accuracy,
+        global_step=args.epochs)
+
+    if args.model_name:
+        # Export the trained model
+        torch.save(net.state_dict(), args.model_name)
+
+        if args.job_dir:
+            # Save the model to GCS
+            data_utils.save_model(args.job_dir, args.model_name)
+    else:
+        print('Accuracy: {:.0f}%'.format(latest_accuracy))
+
+
+def main():
+    # Training settings
+    parser = argparse.ArgumentParser(description='PyTorch Sonar Example')
```

Comment:
nit: create a different method for reading args, as other samples: (Example: get_args()) https://github.com/GoogleCloudPlatform/cloudml-samples/blob/master/boston/tensorflow/keras/trainer/task.py#L28